ocpi-toolkit-2.2.1/src/main/kotlin/com/izivia/ocpi/toolkit/modules/sessions/services/SessionsValidators.kt

```diff
@@ -1,29 +1,30 @@
 package com.izivia.ocpi.toolkit.modules.sessions.services
 
-import com.izivia.ocpi.toolkit.common.validation.*
+import com.izivia.ocpi.toolkit.common.validation.hasMaxLengthOf
+import com.izivia.ocpi.toolkit.common.validation.isCountryCode
+import com.izivia.ocpi.toolkit.common.validation.isCurrencyCode
+import com.izivia.ocpi.toolkit.common.validation.isPrintableAscii
 import com.izivia.ocpi.toolkit.modules.cdr.domain.*
 import com.izivia.ocpi.toolkit.modules.sessions.domain.*
 import com.izivia.ocpi.toolkit.modules.types.Price
 import com.izivia.ocpi.toolkit.modules.types.PricePartial
 import com.izivia.ocpi.toolkit.modules.types.toPartial
-import org.valiktor.functions.isGreaterThan
 import org.valiktor.functions.isGreaterThanOrEqualTo
 import org.valiktor.validate
 import java.math.BigDecimal
-import java.time.Instant
 
 fun SessionPartial.validate(): SessionPartial = validate(this) {
     validate(SessionPartial::countryCode).isCountryCode(caseSensitive = false, alpha2 = true)
     validate(SessionPartial::partyId).isPrintableAscii().hasMaxLengthOf(3)
     validate(SessionPartial::id).isPrintableAscii().hasMaxLengthOf(36)
-    validate(SessionPartial::startDateTime).isGreaterThan(Instant.now())
-    validate(SessionPartial::endDateTime).isGreaterThan(it.startDateTime ?: Instant.now())
+    // startDateTime nothing to validate
+    // endDateTime nothing to validate
```

Comment:
not clear how I can keep the 'greaterThan(startTime)` check, given how this could be a patch, that doesn't include the startTime

Reply:
The only way is to get the actual stored session before the validation 🤷🏻